packages/pwa/app/utils/url.js

```diff
@@ -197,64 +237,97 @@ export const removeQueryParamsFromPath = (path, keys) => {
 }
 
 /**
- * Rebuild the path with locale/site value to the path as url path or url query param
- * based on url config
- * @param {string} url - based url of the output url
- * @param {object} configValues - object that contains values of url param config
- * @return {string} - an output url
+ * Rebuild the path with locale/site values with a given url
+ * The position of those values will based on the url config of your current app configuration.
+ *
+ * @param {string} relativeUrl - the base relative Url to be reconstructed on
+ * @param {object} configValues - object that contains values of url config
+ * @param {Object} [opts] - Options, if there's any.
+ * @param {string[]} opts.disallowParams - URL parameters to remove
+ * @param {object} defaults - place to store default values
+ * @param {object} defaults.defaultLocaleRefs - a list of locale refs of a  site
+ * @return {string} - an output path that has locale and site
  *
  * @example
- * //pwa-kit-config.json
+ * // configuration
  * url {
- *    locale: "query_param"
+ *    locale: "query_param",
+ *    site: "path",
+ *    showDefault: true
  * }
- * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
  *
- * Returns
- *  /women/dresses?locale=en-GB
  *
- *  @example
- * //pwa-kit-config.json
- * url {
- *    locale: "path"
+ * const site = {
+ *   id: 'RefArch',
+ *   alias: 'global'
+ *   l10n: {
+ *     defaultLocale: 'en-GB'
+ *     supportedLocales: [
+ *       {id: 'en-GB', preferCurrency: 'GBP'}
+ *     ]
+ *     // other props
+ *   }
  * }
- * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
- *
- * Returns
- *  /en-GB/women/dresses
+ * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB', site: 'global'}, {site})
+ * => /global/women/dresses?locale=en-GB
  *
  */
-export const buildPathWithUrlConfig = (url, configValues = {}) => {
+export const buildPathWithUrlConfig = (
+    relativeUrl,
+    configValues = {},
+    defaults = {},
+    opts = {}
+) => {
     const urlConfig = getUrlConfig()
-    if (!urlConfig || !Object.values(urlConfig).length) return url
-    if (!Object.values(configValues).length) return url
-    const queryParams = {}
-    const basePathSegments = []
+    const {disallowParams = []} = opts
+    if (!Object.values(configValues).length) return relativeUrl
+    const [pathname, search] = relativeUrl.split('?')
+
+    const params = new URLSearchParams(search)
+    // Remove any disallowed params.
+    if (disallowParams.length) {
+        disallowParams.forEach((param) => {
+            params.delete(param)
+        })
+    }
+
+    const {defaultLocaleRefs = []} = defaults
```

Comment:
Sorry, I'm not sure I understand why `buildPathWithUrlConfig` needs to have `defaults` passed in. What if the function figure out on its own what the defaults are via `getSites`?

Reply:
I'm a little confused as well. @alexvuong maybe you can explain why you need to pass in default values? Would you be able to set the `configValues` to the defaults before calling the function if you know they are not defined?

Reply:
@vmarta The defaults or defaultLocales specific will change depending on the current site. 🤔 . We might be able to solve it without passing the defaults by extracting all default values from the sites returning from`getSites`. 

@bendvc the `configValues` just contain the value to build the url, it does not know whether it is the defaults or not. It has to have something to compare with. As Vincent M mentioned above, we can make the function figure out on its own what the defaults are

Reply:
we had a chat about this. It is possible to resolve the site based on the configValues, then extract the defaultlocales from there. There is no need to pass the `defaults`